src/Orckestra.Composer.Website/UI.Package/Typescript/Composer.Cart/OrderHistory/Services/OrderService.ts

```diff
@@ -27,8 +30,24 @@ module Orckestra.Composer {
                 });
         }
 
-        public saveEditOrder() {
-            return this.orderRepository.saveEditOrder();
+        public saveEditOrder(orderNumber: string) {
+            return this.orderRepository.saveEditOrder(orderNumber).then((result: ICompleteCheckoutResult) => {
+
+                result.IsUpdatedOrder = true;
+                this.cacheProvider.defaultCache.set(this.orderCacheKey, result).done();
+                this.cacheProvider.defaultCache.set(this.orderConfirmationCacheKey, result).done();
+
+                this.eventHub.publish(MyAccountEvents.EditOrderFinished, { data: { orderNumber } });
+
+                this.cartService.invalidateCache();
+                this.cartService.getFreshCart(true);
+
+                if (result.NextStepUrl) {
+                    window.location.href = result.NextStepUrl;
+                }
+
+            })
+                .fail(reason => ErrorHandler.instance().outputError(reason));
```

Comment:
need to define some reaseon key and add texts for it in Errors.
something like 
ErrorHandler.instance().outputError('L_UpdatingOrderFailed')

L_UpdatingOrderFailed
An error occurred while updating your order. Please try again or contact us.